examples/ethereum-tracker/src/contract.rs

```diff
@@ -42,15 +41,24 @@ impl Contract for EthereumTrackerContract {
     async fn instantiate(&mut self, argument: InstantiationArgument) {
         // Validate that the application parameters were configured correctly.
         self.runtime.application_parameters();
-        self.state.argument.set(argument);
-        self.state.last_block.set(0);
-        self.read_initial().await;
+        let InstantiationArgument {
+            ethereum_endpoint,
+            contract_address,
+            start_block,
+        } = argument;
+        let endpoint_and_address = EndpointAndAddress {
+            ethereum_endpoint,
+            contract_address,
+        };
+        self.state.argument.set(endpoint_and_address);
+        self.state.start_block.set(0);
```

Comment:
Shouldn't this be set to `start_block`?

Reply:
It is neutral. This does not change the result because there is no relevant block in the interval `0..start_block`.

Reply:
But doesn't it make the query to the Ethereum node slower, if it has to search through all the earlier blocks rather than just one?

Reply:
Yes, a little. I corrected anyway following your suggestion.